### PR TITLE
Add application icons

### PR DESCRIPTION
**Pull request by jemrobinson**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :vertical_traffic_light: Depends on

n/a

### :arrow_heading_up: Summary

**IMPORTANT: I am confident that merging this will not affect the findings of the ongoing penetration test** 

- Add application icons for
  - Gitea
  - Hedgedoc
  - /data folder
  - /output folder
  - /shared folder

### :closed_umbrella: Related issues

n/a

### :microscope: Tests

Tested on a fresh SRE deployment

Screenshot 2024-08-06 at 21 17 07

### CODE REVIEW

File: data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml [hunk 1]
```diff
@@ -93,6 +93,16 @@ write_files:
...
+  - path: "/root/gitea.url"
+    permissions: "0400"
+    content: |
+      URL=http://{{gitea_hostname}}
+
+  - path: "/root/hedgedoc.url"
+    permissions: "0400"
+    content: |
+      URL=http://{{hedgedoc_hostname}}
+
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:105]: Let's do this a more sensible way

File: data_safe_haven/resources/workspace/ansible/desired_state.yaml [hunk 2]
```diff
@@ -200,6 +201,36 @@
...
+        dest: /usr/share/icons/{{ item }}
+        mode: '0444'
+      loop:
+        - gitea.png
+        - hedgedoc.png
+
+    - name: Copy desktop files
+      ansible.builtin.copy:
+        src: etc/skel/Desktop/{{ item }}
+        dest: /etc/skel/Desktop/{{ item }}
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/ansible/desired_state.yaml]:
```suggestion
        src: "etc/skel/Desktop/{{ item }}"
        dest: "/etc/skel/Desktop/{{ item }}"
```

File: data_safe_haven/resources/workspace/ansible/desired_state.yaml [hunk 3]
```diff
@@ -200,6 +201,36 @@
         - xrdp
         - xrdp-sesman
 
+    - name: Copy icons
+      ansible.builtin.copy:
+        src: usr/share/icons/{{ item }}
+        dest: /usr/share/icons/{{ item }}
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/ansible/desired_state.yaml]:
```suggestion
        src: "usr/local/share/icons/{{ item }}"
        dest: "/usr/local/share/icons/{{ item }}"
```

**Review by JimMadge — DISMISSED** [review]

File: data_safe_haven/resources/workspace/ansible/files/etc/skel/Desktop/hedgedoc.desktop [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Version=1.0
+Type=Link
+Name=HedgeDoc
+Comment=
+Icon=/usr/share/icons/hedgedoc.png
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/ansible/files/etc/skel/Desktop/hedgedoc.desktop]:
```suggestion
Icon=/usr/local/share/icons/hedgedoc.png
```

File: data_safe_haven/resources/workspace/ansible/files/etc/skel/Desktop/gitea.desktop [hunk 5]
```diff
@@ -0,0 +1,6 @@
+[Desktop Entry]
+Version=1.0
+Type=Link
+Name=Gitea
+Comment=
+Icon=/usr/share/icons/gitea.png
```
Comment (JimMadge) [inline data_safe_haven/resources/workspace/ansible/files/etc/skel/Desktop/gitea.desktop]:
```suggestion
Icon=/usr/local/share/icons/gitea.png
```

File: data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:105]: I think we need a general tidy-up of how we pass runtime variables to ansible, but I'm not sure this PR is the right place to put it. I think this is captured in #2075 - do you agree?

Comment (JimMadge) [reply data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:105]: I think I'd like to try it on this PR (I ran out of time yesterday). If it works then follow that pattern elsewhere, if not, merge this as is.

Comment (jemrobinson) [reply data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:105]: Are you still looking into this? I see you made some commits - did they work?

Comment (JimMadge) [reply data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:105]: Yep, testing this now.

**Comment by jemrobinson** [comment]
@JimMadge : These changes feel bigger than the original PR now. Can you start a new PR for them and revert this to `6d0da02`?

Edit: I've made a new branch [`better-ansible-runtime-variables`](https://github.com/alan-turing-institute/data-safe-haven/tree/better-ansible-runtime-variables) for these changes and reverted this branch.

File: data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml:105]: Moved to [`better-ansible-runtime-variables`](https://github.com/alan-turing-institute/data-safe-haven/tree/better-ansible-runtime-variables)

**Review by JimMadge — APPROVED** [review]

**Pull request merged**